### PR TITLE
Floating size, overflow and boundary features and fixes.

### PR DESCRIPTION
**Pull request by xvaara**
# Describe the PR

fix(BDropdown): fix dropup prop ignored
feat(BDropdown): overflow auto and sizing middleware to limit the size to visible viewport
feat(BPopover): overflow auto and sizing middleware to limit the size to visible viewport
feat(BDropdown): add boundaryPadding prop
feat(BPopover): add boundaryPadding prop
feat(BPopover): add boundary prop
fix(BPopover): hide not working

## Small replication

A small replication or video walkthrough can help demonstrate the changes made. This is optional, but can help observe the intended changes. A mentioned issue that contains a replication also works.

## PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix :bug: - `fix(...)`
- [x] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [ ] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [x] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

### CODE REVIEW

File: apps/playground/src/components/Comps/TPopover.vue [hunk 1]
```diff
@@ -60,6 +60,15 @@
           <template #default> <b>Tooltip</b> content </template>
         </BTooltip>
       </BCol>
+      <BCol>
+        <BPopover click :placement="popoverPlacement">
+          <template #title> Click </template>
+          <template #target>
+            <BButton>click longlist</BButton>
+          </template>
+          <div v-for="(_, i) in Array(200)" :key="i">foobar {{ i }}</div>
```
Comment (VividLemon) [inline apps/playground/src/components/Comps/TPopover.vue:69]:
```suggestion
          <div v-for="i in 200" :key="i">foobar {{ i }}</div>
```
Should be valid syntax. Not that I'm super picky on the test environments, just trying to teach you about it if you didn't know about it

File: packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue [hunk 2]
```diff
@@ -218,6 +222,7 @@ const floatingPlacement = computed(() =>
     alignEnd: endBoolean.value,
   })
 )
+const sizeStyles = ref<{maxHeight?: string; maxWidth?: string}>({})
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue:225]:
```suggestion
const sizeStyles = ref<StyleValue>({})
```
This should work better

File: packages/bootstrap-vue-next/src/components/BPopover.vue [hunk 3]
```diff
@@ -192,32 +206,76 @@ const sanitizedContent = computed(() =>
...
 const offsetNumber = useToNumber(() => props.offset ?? NaN)
 
+const boundary = computed<Boundary | undefined>(() =>
+  props.boundary === 'document' || props.boundary === 'viewport' ? undefined : props.boundary
+)
+const rootBoundary = computed<RootBoundary | undefined>(() =>
+  props.boundary === 'document' || props.boundary === 'viewport' ? props.boundary : undefined
+)
+
+const sizeStyles = ref<{maxHeight?: string; maxWidth?: string}>({})
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BPopover.vue:216]: Same as above

**Review by VividLemon — CHANGES_REQUESTED** [review]

File: apps/playground/src/components/Comps/TPopover.vue (hunk 1 above)
Comment (xvaara) [reply apps/playground/src/components/Comps/TPopover.vue:69]: oh, didn't know that, thanks! 😄

File: packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue (hunk 2 above)
Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BDropdown/BDropdown.vue]: That gave me error: Type instantiation is excessively deep and possibly infinite.
CSSProperties worked.

**Pull request merged**